persistence/persistence-mybatis-test/src/test/java/org/cbioportal/persistence/mybatis/PatientMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,177 @@
+        Assert.assertEquals("TCGA-A1-A0SD", result.get(1).getStableId());
+        Assert.assertEquals("TCGA-A1-A0SE", result.get(2).getStableId());
+        Assert.assertEquals("TCGA-A1-A0SF", result.get(3).getStableId());
+        Assert.assertEquals("TCGA-A1-A0SG", result.get(4).getStableId());
+        Assert.assertEquals("TCGA-A1-A0SH", result.get(5).getStableId());
+    }
+
+    @Test
+    public void getMetaPatientsInStudy() throws Exception {
+
+        BaseMeta result = patientMyBatisRepository.getMetaPatientsInStudy("study_tcga_pub");
+
+        Assert.assertEquals((Integer) 14, result.getTotalCount());
+    }
+
+    @Test
+    public void getPatientInStudyNullResult() throws Exception {
+
+        Patient result = patientMyBatisRepository.getPatientInStudy("study_tcga_pub", "invalid_patient");
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void getPatientInStudy() throws Exception {
+
+        Patient patient = patientMyBatisRepository.getPatientInStudy("study_tcga_pub", "TCGA-A1-A0SI");
+
+        Assert.assertEquals((Integer) 7, patient.getInternalId());
+        Assert.assertEquals("TCGA-A1-A0SI", patient.getStableId());
+        Assert.assertEquals((Integer) 1, patient.getCancerStudyId());
+        Assert.assertEquals("study_tcga_pub", patient.getCancerStudyIdentifier());
+        CancerStudy cancerStudy = patient.getCancerStudy();
+        Assert.assertEquals((Integer) 1, cancerStudy.getCancerStudyId());
+        Assert.assertEquals("study_tcga_pub", cancerStudy.getCancerStudyIdentifier());
+        Assert.assertEquals("brca", cancerStudy.getTypeOfCancerId());
+        Assert.assertEquals("Breast Invasive Carcinoma (TCGA, Nature 2012)", cancerStudy.getName());
+        Assert.assertEquals("BRCA (TCGA)", cancerStudy.getShortName());
+        Assert.assertEquals("<a href=\\\"http://cancergenome.nih.gov/\\\">The Cancer Genome Atlas (TCGA)</a> Breast" +
+            " Invasive Carcinoma project. 825 cases.<br><i>Nature 2012.</i> <a href=\\\"http://tcga-data.nci." +
+            "nih.gov/tcga/\\\">Raw data via the TCGA Data Portal</a>.", cancerStudy.getDescription());
+        Assert.assertEquals(true, cancerStudy.getPublicStudy());
+        Assert.assertEquals("23000897", cancerStudy.getPmid());
+        Assert.assertEquals("TCGA, Nature 2012", cancerStudy.getCitation());
+        Assert.assertEquals("SU2C-PI3K;PUBLIC;GDAC", cancerStudy.getGroups());
+        Assert.assertEquals((Integer)0 , cancerStudy.getStatus());
+    }
+
+    @Test
+    public void fetchPatients() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        studyIds.add("study_tcga_pub");
+        List<String> patientIds = new ArrayList<>();
+        patientIds.add("TCGA-A1-A0SB");
+        patientIds.add("TCGA-A1-A0SE");
+
+        List<Patient> result = patientMyBatisRepository.fetchPatients(studyIds, patientIds, "SUMMARY");
```

Comment:
This pairing of arguments by array index feels wrong .. but I guess doing it this way allows mapping to the same handler in the mybatis mapper, rather than having one handler for a list of (study,patient) pairing objects .. and a different handler for a simple list of study identifiers (patientIds=null).